pandas/plotting/_converter.py

```diff
@@ -246,7 +246,9 @@ def _convert_1d(values, units, axis):
             return values.asfreq(axis.freq)._ndarray_values
         if isinstance(values, Index):
             return values.map(lambda x: get_datevalue(x, axis.freq))
-        if is_period_arraylike(values):
+        if lib.infer_dtype(values) == 'period':
```

Comment:
could be elif if you want

Reply:
Actually had an elif originally, but changed it to an `if` to be consistent with the rest (I could change them all to `elif` if you want)

Reply:
prob more idiomatic as i think they all return.